slither/printers/summary/cheatcodes.py

```diff
@@ -0,0 +1,73 @@
+"""
+    Cheatcode printer
+
+    This printer prints the usage of cheatcode in the code.
+"""
+from slither.printers.abstract_printer import AbstractPrinter
+from slither.slithir.operations import HighLevelCall
+from slither.utils import output
+
+
+class CheatcodePrinter(AbstractPrinter):
+
+    ARGUMENT = "cheatcode"
+
+    HELP = """
+        Print the usage of (Foundry) cheatcodes in the code.
+        For the complete list of Cheatcodes, see https://book.getfoundry.sh/cheatcodes/
+    """
+
+    WIKI = "https://github.com/trailofbits/slither/wiki/Printer-documentation#cheatcode"
+
+    def output(self, filename: str) -> output.Output:
+
+        info: str = ""
+
+        try:
+            vm = self.slither.get_contract_from_name("Vm").pop()
+        except IndexError:
+            return output.Output("No contract named VM found")
+
+        for contract in self.slither.contracts_derived:
+            # Check that the IS_TEST variable is set. (Only works for Foundry)
+            is_test_var = contract.variables_as_dict.get("IS_TEST", None)
+            is_test = False
+            if is_test_var is not None:
+                try:
+                    is_test = is_test_var.expression.value == "true"
+                except AttributeError:
+                    pass
+
+            if not is_test:
+                continue
+
+            found_contract: bool = False
+            contract_info: str = ""
+            for func in contract.functions_declared:
+                function_info = f"\t{func}\n"
+                found_function: bool = False
+                for node in func.nodes:
+                    for op in node.all_slithir_operations():
+                        if (
+                            isinstance(op, HighLevelCall)
+                            and op.function.contract == vm
+                            and op.function.visibility == "external"
+                        ):
+                            found_function = True
+                            function_info += (
+                                f"\t\tL{node.source_mapping.lines}: {op.function.name}\n"
+                            )
```

Comment:
Can you use `to_detailed_str()` here please? And maybe show the source mapping content?

Reply:
New output is like this

```
INFO:Printers:CounterTest (test/Counter.t.sol)
	setUp
		deal - (test/Counter.t.sol#21 (9 - 32)
		vm.deal(alice,1000000000000000000)

		deal - (test/Counter.t.sol#22 (9 - 30)
		vm.deal(bob,2000000000000000000)

	testIncrement
		prank - (test/Counter.t.sol#28 (9 - 24)
		vm.prank(alice)

		assertEq - (test/Counter.t.sol#30 (9 - 38)
		assertEq(counter.number(),1)

		prank - (test/Counter.t.sol#32 (9 - 22)
		vm.prank(bob)

		assertEq - (test/Counter.t.sol#34 (9 - 38)
		assertEq(counter.number(),2)
```

I'm open to any suggestion to enhance it.